sim-lib/src/random_activity.rs

```diff
@@ -266,6 +282,18 @@ impl Display for RandomPaymentActivity {
     }
 }
 
+/// Random number generator that takes an optional seed with which deterministic
+/// behaviour can be simulated.
+fn sim_rng(seed: Option<u64>) -> Box<dyn RngCore> {
```

Comment:
We can "just in time" create a `thread_rng` because it gets fresh entropy on each instantiation, but if we do this with a set seed, we'll always get the same value because we're creating the same fixed seed generator each time. This means that we'll always make the same choices every time we sample, which isn't what we want with a fixed seed. 

When I first implemented this, I tried to keep a single instance of `thread_rng` in the struct and it got really hairy because it's not `Send`. I think that a reasonable approach would be to replace `seed` in the various structs with `seeded_rng: Option<Box<dyn RngCore + Send>>` which is `Some` if we have a set seed and `None` otherwise. Should be relatively painless, and then we can use `unwrap_or` to use the seeded rng or just `thread_rng` inline where we need it?

See [test demonstrating issue](https://github.com/bitcoin-dev-project/sim-ln/files/15379642/test_rng.patch), sorry that I didn't pick this up on first pass!